modules/xfem/src/userobjects/RectangleCutUserObject.C

```diff
@@ -0,0 +1,82 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "RectangleCutUserObject.h"
+
+// MOOSE includes
+#include "MooseError.h"
+
+// XFEM includes
+#include "XFEMFuncs.h"
+//#include "EFAFuncs.h"
+
+template <>
+InputParameters
+validParams<RectangleCutUserObject>()
+{
+  // Get input parameters from parent class
+  InputParameters params = validParams<GeometricCut3DUserObject>();
+
+  // Add required parameters
+  params.addRequiredParam<std::vector<Real>>("cut_data",
+                                             "Vector of Real values providing cut information");
+  // Class description
+  params.addClassDescription("Creates a UserObject for planar cuts on 3D meshes for XFEM");
+  // Return the parameters
+  return params;
+}
+
+RectangleCutUserObject::RectangleCutUserObject(const InputParameters & parameters)
+  : GeometricCut3DUserObject(parameters), _cut_data(getParam<std::vector<Real>>("cut_data"))
+{
+  // Throw error if length of cut_data is incorrect
+  if (_cut_data.size() != 12)
+    mooseError("Length of RectangleCutUserObject cut_data must be 12");
+
+  // Assign cut_data to vars used to construct cuts
+  _vertices.push_back(Point(_cut_data[0], _cut_data[1], _cut_data[2]));
+  _vertices.push_back(Point(_cut_data[3], _cut_data[4], _cut_data[5]));
+  _vertices.push_back(Point(_cut_data[6], _cut_data[7], _cut_data[8]));
+  _vertices.push_back(Point(_cut_data[9], _cut_data[10], _cut_data[11]));
+
+  for (unsigned int i = 0; i < 4; ++i)
+    _center += _vertices[i];
+  _center *= 0.25;
+
+  for (unsigned int i = 0; i < 4; ++i)
+  {
+    unsigned int iplus1(i < 3 ? i + 1 : 0);
+    Point ray1 = _vertices[i] - _center;
+    Point ray2 = _vertices[iplus1] - _center;
+    _normal += ray1.cross(ray2);
+  }
+  _normal *= 0.25;
+  Xfem::normalizePoint(_normal);
+}
+
+RectangleCutUserObject::~RectangleCutUserObject() {}
+
+bool
+RectangleCutUserObject::isInsideCutPlane(Point p) const
+{
+  bool inside = false;
+  unsigned int counter = 0;
+  for (unsigned int i = 0; i < 4; ++i)
+  {
+    unsigned int iplus1 = (i < 3 ? i + 1 : 0);
+    Point middle2p = p - 0.5 * (_vertices[i] + _vertices[iplus1]);
+    const Point side_tang = _vertices[iplus1] - _vertices[i];
+    Point side_norm = side_tang.cross(_normal);
+    Xfem::normalizePoint(middle2p);
+    Xfem::normalizePoint(side_norm);
+    if (middle2p * side_norm <= 0.0)
+      counter += 1;
+  }
+  if (counter == 4)
+    inside = true;
+  return inside;
+}
```

Comment:
This class seems to be loaded with magic numbers (constants) Can you use named constants instead where appropriate?